pkg/kclient/ingress_test.go

```diff
@@ -0,0 +1,236 @@
+package kclient
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	extensionsv1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ktesting "k8s.io/client-go/testing"
+)
+
+func TestCreateIngress(t *testing.T) {
+
+	tests := []struct {
```

Comment:
since the function takes label selector as an input, the struct should at least contain the below fields

```
type args struct {
		labelSelector string
}
tests := []struct {
                args    args
		wantIngress   []extensionsv1.Ingress
		wantErr bool
}
```
the ingressName should be not be used and should instead be a part of ingress as the length of the returned list can be greater than 1